api/src/libs/gh-utils.js

```diff
@@ -0,0 +1,88 @@
+//
+// Realm-o-Matic
+//
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Created by Shelly Xue Han
+//
+
+'use strict';
+
+import _ from 'lodash';
+import { errorWithCode } from '@bcgov/nodejs-common-utils';
+import { realmSchema } from '../constants';
+import { validateSchema } from './utils';
+import shared from './shared';
+import config from '../config';
+
+// Normalize the github issue:
+export const normalizeIssue = issueBody => {
+  try {
+    const issueContent = issueBody.body;
+    const issueId = issueBody.number;
+    if (!issueContent || !issueId) throw Error('Invalid GitHub issue.');
+    const realmInfo = JSON.parse(issueContent.replace(/\r?\n|\r/g, ''));
+    const { isValid, payload } = validateSchema(realmInfo, realmSchema);
+
+    if (!isValid) throw Error(payload);
+    return { issueId, realm: payload };
+  } catch (err) {
+    throw Error(err);
+  }
+};
+
+// Get the array of normalize github issues:
+export const normalizeIssues = data => {
+  try {
+    if (!_.isEmpty(data)) {
+      return _.isArray(data) ? data.map(i => normalizeIssue(i)) : [normalizeIssue(data)];
+    }
+    throw errorWithCode('No issues found.', 404);
+  } catch (err) {
+    throw Error(err.message);
```

Comment:
I think you just want to re-throw `err` here. I often use the catch block as an opportunity to log before regrowing the error. If you don't it won't have a `code` so the catch-all will just fill in 500.

For example:

```
    } catch (err) {
      const message = `Unable to remove bla ${blaId} from order ${orderId}`;
      logger.error(`${message}, err = ${err.message}`);

      throw errorWithCode(message, err.code ? err.code : 500);
    }
```

Reply:
PS: `logger` comes from `'@bcgov/nodejs-common-utils'`.